packages/analytics/src/client.ts

```diff
@@ -0,0 +1,36 @@
+import { AnalyticsClientConfig, AnalyticsClientInterface, EventProperties, JsonMap } from './types';
+
+export type AnalyticsClientOptions = Pick<
+  AnalyticsClientConfig,
+  'defaultProperties' | 'defaultTraits'
+>;
+
+export class AnalyticsClient<T extends AnalyticsClientInterface> {
+  constructor(
+    private readonly client: T,
+    private readonly options: AnalyticsClientOptions
+  ) {
+    this.client = client;
+    this.options = options;
```

Comment:
private keyword of the class already defines the parameter to the class, so these lines are redundant afaik?
```suggestion
```